paddle/fluid/framework/new_executor/new_executor_defs.cc

```diff
@@ -693,6 +693,29 @@ const std::vector<size_t>& Instruction::GCCheckVars() const {
   return gc_check_var_list_;
 }
 
+void Instruction::InferNeedStreamSyncVars() {
+  auto ParseIndexFromMap = [](
+      std::map<std::string, std::vector<int>>& index_map,
+      std::vector<size_t>* out_vec) {
+    for (auto& pair : index_map) {
+      std::vector<int>& indexes = pair.second;
+      out_vec->insert(out_vec->end(), indexes.begin(), indexes.end());
+    }
+  };
+  ParseIndexFromMap(op_func_node_.input_index, &need_stream_sync_var_list_);
+  // ParseIndexFromMap(op_func_node_.output_index, &need_stream_sync_var_list_);
+
+  // NOTE(Ruibiao): A simple and general approach is to record all input and
+  // output vars of the instruction. However, we ignore the output vars here,
+  // because the output of both H2D and D2H must be needed and will not be GC
```

Comment:
What's the reason of  `will not be GC`, it should be also added in the comments.

Reply:
The comments have been modified, see [LINK](https://github.com/PaddlePaddle/Paddle/blob/fb0cb883cfbf8f886f3c22c2a60bee25831247a5/paddle/fluid/framework/new_executor/interpretercore_garbage_collector.cc#L132)